paper-menu.html

```diff
@@ -0,0 +1,330 @@
+      /***********************************************************************/
+
+      /**
+       * If you want to use the attribute value of an element for `selected` instead of the index,
+       * set this to the name of the attribute.
+       *
+       * @attribute attrForSelected
+       * @type String
+       */
+      attrForSelected: {
+        type: String
+      },
+
+      /**
+       * If true, multiple selections are allowed.
+       *
+       * @attribute multi
+       * @type Boolean
+       * @default false
+       */
+      multi: {
+        observer: 'multiChanged',
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * Gets or sets the selected element. The default is to use the index of the item.  In
+       * multi-selection this should be an array of values.
+       *
+       * @attribute selected
+       * @type String|Array
+       */
+      selected: {
+        notify: true,
+        observer: 'selectedChanged',
+        type: String
+      },
+
+      /**
+       * Returns the currently selected item. In multi-selection this returns an array of
+       * selected items.
+       *
+       * @attribute selectedItem
+       * @type Object|Array
+       */
+      selectedItem: {
+        notify: true,
+        observer: 'selectedItemChanged',
+        readOnly: true,
+        type: Object
+      },
+
+      /**
+       * The event that would be fired from the item to indicate it is being selected.  Set this
+       * to empty string or null if you don't want to listen for any events.
+       *
+       * @attribute activateEvent
+       * @type String
+       * @default 'click'
+       */
+      activateEvent: {
+        observer: 'activateEventChanged',
+        type: String,
+        value: 'click'
+      },
+
+      /**
+       * If this is set, only items with local name that matches the `selectable` are selectable.
+       *
+       * @attribute selectable
+       * @type String
+       */
+      selectable: {
+        type: String
+      }
+
+    },
+
+    hostAttributes: {
+      'role': 'menu',
+      'tabindex': '0'
+    },
+
+    listeners: {
+      'focus': 'onFocus',
+      'keydown': 'onKeydown'
+    },
+
+    created: function() {
+      this._bindActivateHandler = this.activateHandler.bind(this);
+    },
+
+    attached: function() {
+      this.selectableAttached();
+    },
+
+    detached: function() {
+      this.selectableDetached();
+      this.removeListener(this.activateEvent);
+    },
+
+    addListener: function(eventName) {
+      if (eventName) {
+        this.addEventListener(eventName, this._bindActivateHandler);
+      }
+    },
+
+    removeListener: function(eventName) {
+      if (eventName) {
+        this.removeEventListener(eventName, this._bindActivateHandler);
+      }
+    },
+
+    activateEventChanged: function(eventName, old) {
+      this.removeListener(old);
+      this.addListener(eventName);
+    },
+
+    focusedItemChanged: function(focusedItem, old) {
```

Comment:
It would be cool to generalize this "traveling tabindex" behavior. No action required, just observing.